src/distilabel/steps/filtering/embedding.py

```diff
@@ -0,0 +1,179 @@
+                            -0.07325472251663565,
+                            -0.06058678419516501,
+                            -0.02100326928586996,
+                            -0.013462744792362657,
+                            0.027368447064244242,
+                            -0.003916070100455717,
+                            0.01243614518480423,
+                        ],
+                        "nn_indices": [0, 2],
+                        "nn_scores": [
+                            0.7552462220191956,
+                            0.7261884808540344,
+                        ],
+                    },
+                    {
+                        "persona": "A classical guitar teacher or instructor, likely with experience teaching beginners, who focuses on breaking down complex music notation into understandable steps for their students.",
+                        "embedding": [
+                            -0.01630817942328242,
+                            -0.023760151552345232,
+                            -0.014249650090627883,
+                            -0.005713686451446624,
+                            -0.016033059279131567,
+                            0.0071440908501058786,
+                            -0.05691099643425161,
+                            0.01597412704817784,
+                        ],
+                        "nn_indices": [1, 2],
+                        "nn_scores": [
+                            0.8107735514640808,
+                            0.7172299027442932,
+                        ],
+                    },
+                ],
+                batch_size=batch_size,
+            )
+            # In general you should do something like this before the deduplication step, to obtain the
+            # `nn_indices` and `nn_scores`. In this case the embeddings are already normalized, so there's
+            # no need for it.
+            # nn = FaissNearestNeighbour(
+            #     k=30,
+            #     metric_type=faiss.METRIC_INNER_PRODUCT,
+            #     search_batch_size=50,
+            #     train_size=len(dataset),              # The number of embeddings to use for training
+            #     string_factory="IVF300_HNSW32,Flat"   # To use an index (optional, maybe required for big datasets)
+            # )
+            # Read more about the `string_factory` here:
+            # https://github.com/facebookresearch/faiss/wiki/Guidelines-to-choose-an-index
+
+            embedding_dedup = EmbeddingDedup(
+                threshold=0.8,
+                input_batch_size=batch_size,
+            )
+
+            data >> embedding_dedup
+
+        if __name__ == "__main__":
+            distiset = pipeline.run(use_cache=False)
+            ds = distiset["default"]["train"]
+            # Filter out the duplicates
+            ds_dedup = ds.filter(lambda x: x["keep_row_after_embedding_filtering"])
+        ```
+    """
+
+    threshold: float = 0.9
```

Comment:
do you think it would be good to have this as a `RuntimeParameter`?

Reply:
Yes, will let the option too